src/sqlfluff/rules/layout/LT01.py

```diff
@@ -29,14 +33,20 @@ class Rule_L039(BaseRule):
         FROM foo
     """
 
-    groups = ("all", "core")
+    name = "layout.spacing"
+    # NOTE: This rule combines the following legacy rules:
+    # - L001: Trailing Whitespace
+    # - L039: Unnecessary Whitespace
+    # TODO: Potentially more
+    aliases = ("L001", "L039")
+    groups = ("all", "core", "layout")
     crawl_behaviour = RootOnlyCrawler()
     is_fix_compatible = True
 
     def _eval(self, context: RuleContext) -> Optional[List[LintResult]]:
         """Unnecessary whitespace."""
         sequence = ReflowSequence.from_root(context.segment, config=context.config)
-        results = sequence.respace(filter="inline").get_results()
+        results = sequence.respace().get_results()
 
         # For now, respace rules are separate for creation and reduction.
         # That shouldn't be true in future.
```

Comment:
Is this comment still relevant?

Reply:
It is _for now_. But I'd like to break that constraint in the next PR by bringing in the other spacing rules that do create whitespace when it's missing. That one could reduce the number of rules significantly.